extensions/bitwarden/src/hooks.ts

```diff
@@ -0,0 +1,43 @@
+import { getLocalStorageItem, showToast, ToastStyle, removeLocalStorageItem, setLocalStorageItem } from "@raycast/api";
+import { useState, useEffect } from "react";
+import { BitwardenApi } from "./api";
+
+const LocalStorageSessionKey = "sessionToken"
+export function useBitwarden(bitwardenApi: BitwardenApi): [string | null | undefined, (sessionToken: string | null) => void] {
+  const [sessionToken, setSessionToken] = useState<string | null>();
+
+  useEffect(() => {
+    async function getSessionToken() {
+      const sessionToken = await getLocalStorageItem<string>(LocalStorageSessionKey);
+
+      const status = await bitwardenApi.getVaultStatus(sessionToken)
+
+      if (status === "unlocked") setSessionToken(sessionToken);
+      else if (status === "locked") setSessionToken(null);
+      else if (status === "unauthenticated") {
+        try {
+          const toast = await showToast(ToastStyle.Animated, "Login in...", "It may takes some time");
+          await bitwardenApi.login()
+          toast.hide();
+          setSessionToken(null);
+        } catch (error) {
+          showToast(ToastStyle.Failure, "An error occurred during login!", "Please check your crendentials");
+        }
+      }
+    }
+    getSessionToken();
```

Comment:
Is this the only place this function is called? Does it need to be separately defined above? Can we just call that code inline instead?